evm/vm/state.py

```diff
@@ -407,24 +369,32 @@ def get_uncle_reward(block_number, uncle):
     def get_nephew_reward(cls):
         raise NotImplementedError("Must be implemented by subclasses")
 
+    @abstractmethod
+    def execute_transaction(self):
+        raise NotImplementedError()
+
 
 class BaseTransactionExecutor(metaclass=ABCMeta):
-    def execute_transaction(self, transaction):
-        """
-        Execute the transaction in the vm.
-        """
-        message = self.run_pre_computation(transaction)
-        computation = self.run_computation(transaction, message)
-        return self.run_post_computation(transaction, computation)
+    def __init__(self, transaction, vm_state):
+        self.transaction = transaction
+        self.transaction_context = vm_state.get_transaction_context_class()(
+            gas_price=self.transaction.gas_price,
+            origin=self.transaction.sender,
+            data=self.transaction.data,
+        )
+        self.vm_state = vm_state
+
+        self.message = self.build_transaction_message()
+        self.computation = self.finalize_computation(self.build_computation())
 
     @abstractmethod
-    def run_pre_computation(self, transaction):
+    def build_transaction_message(self):
```

Comment:
This name feels slightly off.  The `message` object is not really the *transaction* message, but the *evm* message so this might be better named `build_evm_message` or even just `build_message.`